### ISSUE TITLE
authorization issues still here!

### ISSUE THREAD

**Issue by odenisenko** [tchellomello/python-ring-doorbell#121]
2019-02-28 08:19:03,253 - doorbell - ERROR - No handlers could be found for logger "ring_doorbell"
2019-02-28 08:19:03,308 - doorbell - ERROR - ('Connection aborted.', error(104, 'Connection reset by peer'))
2019-02-28 08:20:03,357 - doorbell - INFO - Trying to connect to ring service...
2019-02-28 08:20:06,726 - doorbell - ERROR - 401 Client Error: Unauthorized for url: https://api.ring.com/clients_api/session
2019-02-28 08:21:06,785 - doorbell - INFO - Trying to connect to ring service...
2019-02-28 08:21:09,608 - doorbell - ERROR - 401 Client Error: Unauthorized for url: https://api.ring.com/clients_api/session
2019-02-28 08:22:09,611 - doorbell - INFO - Trying to connect to ring service...

**Comment by odenisenko** [comment]
after restart connected OK!

**Comment by odenisenko** [comment]
my code
```
while True:
      try:
           logger.info("Trying to connect to ring service...")
           myring = Ring(ringUsername, ringPassword, debug=True, reuse_session=False)
           if myring.is_connected==False:
              logger.error("Ring Connection Not Successful ".format(myring.is_connected))
              break
           else:
              logger.info("Ring Connection Successful")

           while True:
                 if myring.is_connected==False:
                    logger.error("Ring Connection broken ".format(myring.is_connected))
                    break

                 doorbell = myring.doorbells[0]
                 data = doorbell.check_alerts()
                 if data == True:
                    print(data)
                 sleep(13)

      except Exception as d:
             logger.error(d)
             time.sleep(60)
```

**Comment by odenisenko** [comment]
UPD:
after investigation, we have 2 issues
1) when we got something wrong (401 for example0 from API. our the core can't restore correctly session

```
2019-03-12 10:04:00,112 - ring_doorbell - DEBUG - running query loop 0
2019-03-12 10:04:00,254 - ring_doorbell - DEBUG - _query 1 ret 401
2019-03-12 10:04:02,184 - ring_doorbell - ERROR - Failed: 401 Client Error: Unauthorized for url: https://api.ring.com/clients_api/session
2019-03-12 10:04:02,184 - ring_doorbell - INFO - Trying to connect to ring service...
2019-03-12 10:04:02,185 - ring_doorbell - DEBUG - Querying https://api.ring.com/clients_api/ring_devices
2019-03-12 10:04:02,185 - ring_doorbell - DEBUG - Not connected. Refreshing token...
2019-03-12 10:04:04,679 - ring_doorbell - ERROR - Failed: 401 Client Error: Unauthorized for url: https://api.ring.com/clients_api/session
2019-03-12 10:04:04,681 - ring_doorbell - INFO - Trying to connect to ring service...
````
2) https://api.ring.com/clients_api/snapshots/timestamps end now throw 
2019-03-12 10:04:00,112 - ring_doorbell - DEBUG - Querying https://api.ring.com/clients_api/snapshots/timestamps
```
2019-03-12 10:04:00,112 - ring_doorbell - DEBUG - running query loop 0
2019-03-12 10:04:00,254 - ring_doorbell - DEBUG - _query 1 ret 401
```

**Comment by mandusm** [comment]
related: https://github.com/tchellomello/python-ring-doorbell/issues/142

**Comment by tchellomello** [comment]
Closed and fixed by #143
